app/src/main/java/uk/gov/onelogin/tokens/usecases/GetPersistentId.kt

```diff
@@ -0,0 +1,24 @@
+package uk.gov.onelogin.tokens.usecases
+
+import javax.inject.Inject
+import uk.gov.onelogin.login.usecase.extractPersistentIdFromIdToken
+import uk.gov.onelogin.repositiories.TokenRepository
+
+fun interface GetPersistentId {
+    /**
+     * Use case to get the persistent id from the id token
+     *
+     * @return persistent id as a string or null if it fails to retrieve it
+     */
+    operator fun invoke(): String?
+}
+
+class GetPersistentIdImpl @Inject constructor(
+    val tokenRepository: TokenRepository
+) : GetPersistentId {
+    override fun invoke(): String? {
+        val idToken: String = tokenRepository.getTokenResponse()?.idToken ?: return null
```

Comment:
We might need to know if it fails to retrieve/extract the `persistentSessionID`?

Reply:
Well it will just be null if it fails to retrieve, this is the pattern for retrieving the email for example. It also allows for missing id token _or_ missing persistent id to fail in the same way